it/src/main/java/org/corfudb/universe/group/cluster/CorfuCluster.java

```diff
@@ -9,8 +11,38 @@
 
     /**
      * Provides a corfu client running on local machine
+     *
      * @return local corfu client
      */
     LocalCorfuClient getLocalCorfuClient();
 
+    /**
+     * Find a corfu server by index in the cluster:
+     * - get all corfu servers in the cluster
+     * - skip particular number of servers in the cluster according to index offset
+     * - extract first server from the list
+     *
+     * @param index corfu server position
+     * @return a corfu server
+     */
+    default CorfuServer getServerByIndex(int index) {
+        return nodes()
+                .values()
+                .stream()
+                .skip(index)
+                .findFirst()
```

Comment:
You can just use `return (CorfuServer) nodes().values().asList().get(index)` since .values() is already sorted by key.  This is more clear.

Reply:
Then I have to check if the value is null and throw an exception, in my implementation it's already covered

Reply:
Then I have to check if the value is null and throw an exception, in my implementation it's already covered